src/core/mac/link_raw.cpp

```diff
@@ -51,17 +51,27 @@ namespace Mac {
 
 LinkRaw::LinkRaw(Instance &aInstance)
     : InstanceLocator(aInstance)
-    , mReceiveChannel(OPENTHREAD_CONFIG_DEFAULT_CHANNEL)
-    , mPanId(kPanIdBroadcast)
-    , mReceiveDoneCallback(nullptr)
-    , mTransmitDoneCallback(nullptr)
-    , mEnergyScanDoneCallback(nullptr)
 #if OPENTHREAD_RADIO
     , mSubMac(aInstance)
 #elif OPENTHREAD_CONFIG_LINK_RAW_ENABLE
     , mSubMac(aInstance.Get<SubMac>())
 #endif
 {
+    Init();
+}
+
+void LinkRaw::Init(void)
+{
+    mEnergyScanDoneCallback = nullptr;
+    mTransmitDoneCallback   = nullptr;
+    mReceiveDoneCallback    = nullptr;
+
+    mReceiveChannel      = OPENTHREAD_CONFIG_DEFAULT_CHANNEL;
+    mPanId               = kPanIdBroadcast;
+    mReceiveDoneCallback = nullptr;
```

Comment:
When the `Instance::RadioStackReset()` calls the `mLinkRaw.Init()` , why we need to reset the callback pointers `mReceiveDoneCallback`? Do we expect the up layer to set the callback pointer again after the the `RadioStackReset()` is called?

Reply:
Because `LinkRaw` is using the callback to indicate if it's enabled. I think it's better to use a separate bool to store the enabled state but we can do it in a separate PR.

Currently the upper layer will set the callback regardless of what the lower layer does.